helpers.py

```diff
@@ -76,12 +84,17 @@ def _pages_this_day():
 
 def _random_pages(limit):
     page_len = models.Page.objects.count()
-    indices = random.sample(xrange(page_len), limit)
-    pages = []
-    for index in indices:
-        pages.append(models.Page.objects.all()[index])
-
-    return map(_get_page_by_object, pages)
+    if page_len:
+        indices = ([i for i in xrange(page_len)]
+                    if page_len < limit
+                    else random.sample(xrange(page_len), limit))
+        pages = []
+        for index in indices:
+            pages.append(models.Page.objects.all()[index])
```

Comment:
I'm curious if writing this with the call to `models.Page.objects.all()` inside the `for` loop is inefficient and executing the same query with a large result set for each iteration. Django may be caching it behind the scenes for us, but I'd want to know for sure.